spyder_kernels/customize/code_runner.py

```diff
@@ -0,0 +1,593 @@
+
+            try:
+                if __umr__.has_cython:
+                    # Cython files
+                    with io.open(filename, encoding="utf-8") as f:
+                        self.shell.run_cell_magic("cython", "", f.read())
+                else:
+                    self._exec_code(
+                        file_code,
+                        filename,
+                        ns_globals,
+                        ns_locals,
+                        post_mortem=post_mortem,
+                        exec_fun=exec_fun,
+                        capture_last_expression=False,
+                        global_warning=not current_namespace,
+                    )
+            finally:
+                sys.argv = [""]
+
+    def _exec_cell(
+        self,
+        cell_id,
+        filename=None,
+        post_mortem=False,
+        exec_fun=None,
+        canonic_filename=None,
+        context_locals=None,
+        context_globals=None,
+    ):
+        """
+        Execute a code cell.
+        """
+        try:
+            # Get code from spyder
+            cell_code = frontend_request(blocking=True).run_cell(cell_id, filename)
+        except Exception:
+            print(
+                "This command failed to be executed because an error occurred"
+                " while trying to get the cell code from Spyder's"
+                " editor. The error was:\n\n"
+            )
+            self.shell.showtraceback(exception_only=True)
+            return
+
+        if not cell_code or cell_code.strip() == "":
+            print("Nothing to execute, this cell is empty.\n")
+            return
+
+        # Trigger `post_execute` to exit the additional pre-execution.
+        # See Spyder PR #7310.
+        self.shell.events.trigger("post_execute")
+        file_code = self._get_file_code(filename, save_all=False)
+
+        # Here the remote filename has been used. It must now be valid locally.
+        filename = canonic_filename
+
+        with NamespaceManager(
+            self.shell,
+            filename,
+            current_namespace=True,
+            file_code=file_code,
+            context_locals=context_locals,
+            context_globals=context_globals
+        ) as (ns_globals, ns_locals):
+            return self._exec_code(
+                cell_code,
+                filename,
+                ns_globals,
+                ns_locals,
+                post_mortem=post_mortem,
+                exec_fun=exec_fun,
+                capture_last_expression=True,
+            )
+
+    def _get_current_file_name(self):
+        """Get the current editor file name."""
+        try:
+            return frontend_request(blocking=True).current_filename()
+        except Exception:
+            print(
+                "This command failed to be executed because an error occurred"
+                " while trying to get the current file name from Spyder's"
+                " editor. The error was:\n\n"
```

Comment:
```suggestion
                "This command failed to be executed because an error occurred "
                "while trying to get the current file name from Spyder's editor."
                "The error was:\n\n"
```